library/aes.c

```diff
@@ -675,6 +679,13 @@ int mbedtls_aes_setkey_enc( mbedtls_aes_context *ctx, const unsigned char *key,
 int mbedtls_aes_setkey_dec( mbedtls_aes_context *ctx, const unsigned char *key,
                     unsigned int keybits )
 {
+#if defined(MBEDTLS_AES_ONLY_ENCRYPT)
```

Comment:
Minor - I can see a conscious design choice has been made here, but my choice would have been to remove the entire function and force a linker error should anyone try to use the function. That may have had unintended consequences, but should someone misconfigure the library the error will be very obvious. 

By leaving this stub in, it will compile but fail in execution, which I think is the wrong choice.

However - for `baremetal` I can accept the PR as it is, but for upstreaming I'd like to revisit this choice.

cc: @mpg